### DISCUSSION TITLE
[ Feature Request ] - Ability to re-use Field Groups across Sections/Field Layouts.

### DISCUSSION THREAD

**Discussion by kevindeleon** [craftcms/cms#10501] [Ideas]
Not exactly sure how to word the title of this but basically I would love to see Craft employ a system similar to the Advanced Custom Fields (ACF) plugin for Wordpress in which you can have "field groups" that act like containers that can then be added to pages/post types via the plugin admin based on the content type. This is especially useful when your site has a lot of Singles or you are often creating new Content Types (Channels, Structures).

Example: let's say I have a `Hero Fields` field group for creating a hero banner at the top of pages/posts -- and in that hero group, I have multiple fields like `hero title`, `hero tagline`, `hero image`, etc.

In Craft, I can have a group of fields called `Hero Fields` -- and every time I create a new Channel or Single or Structure... I have to go into Entry Type/Field Layout and create a tab, and then drag those fields to that tab over and over again for each new "content type" I create.

In Wordpress, using ACF, I would create the above stated field group and add the needed fields. Then, I would simply go into the settings in ACF for the particular field group and choose "Show this field group for the following content types" and then check off the ones I want - "posts, pages, recipes, team members, etc..." -- Those content types would automatically have that "tab" with all the needed fields in the "create/edit" screen without the need to repeat the same steps over and over.

As you can see, this is a powerful way to not only re-use fields...but to quickly setup new sections without having to create new tabs every time, and then drag and drop 10 fields or whatever.

**Comment by simonkuran** [comment]
I would suggest that the headings for each field group can be dragged _into_ a tab group in the entry type editor to add the field to a particular tab, or _on to_ the background of the entry type editor to create a new tag group. If I'm remembering correctly this was an option before the Craft 3.5 field layout designer.
![Screen Shot 2022-02-09 at 9 35 01 AM](https://user-images.githubusercontent.com/41410727/153128328-a4eedff3-1dcd-4395-bb74-ec969946c286.png)

**Reply by kevindeleon** [reply]
Yeah I think that would be a step in the right direction and would be much more convenient than it is now...however, being able to create a field group, and then assign it to multiple Entry Types, or "all singles" at one time would be much more convenient than having to go into each one and edit the field layout. I'm sure that's a big ask, but it saves SO many hours when setting up large sites with lots of channels (content types in Wordpress) and singles (pages in Wordpress) -- and you have a decent sized set of common fields.

**Comment by tremby** [comment]
You can sort of do this already with Matrix: make a new Matrix field, define exactly one block type and put the fields you want in that, and then depending whether you want to require the group, and whether you want to allow repeating or not, possibly configure the Matrix to have a minimum and maximum number of blocks.

But the challenges with this are that the UI is clunky (even though there's only one block type option, you get nested headings for the matrix itself, then for the block type), and that it's currently impossible to nest a Matrix within another Matrix and so you wouldn't be able to reuse this set of fields in a content builder Matrix.

What I'd like is either for this use case to get its own treatment (a "group" field type, perhaps), or for Matrix to support this use case better, i.e.
- recognize when there's exactly one block type and change the UI appropriately
- allow nesting Matrix within Matrix
